### ISSUE TITLE
Bug: Prize Catch when mooch 2 is NOT available.

### ISSUE THREAD

**Issue by DevThew** [InitialDet/AutoHook#103]
I don't think this option is working at all, since it always use no matter if its ticked or not.

Expected: Only use Prize catch if my mooch 2 is on cooldown.

Results: It's being used if i have mooch2 available or not.

Tested both ticked and unticked options, mooch 2 use is ticked as requested by the description.

**Comment by resech** [comment]
I'm seeing the same thing at the moment. It looks like the code is checking for a return of 0:

```c#
    // status 0 == available to cast? not sure but it seems to be
    // Also make sure its the skill is not on cooldown (mainly for mooch2)
    public static unsafe bool ActionAvailable(uint id, ActionType actionType = ActionType.Action)
    {
        if (actionType == ActionType.Item)
            return true;

        return ActionManager->GetActionStatus(actionType, id) == 0 && !FFXIVClientStructs.FFXIV.Client.Game.ActionManager.Instance()->IsRecastTimerActive(ActionType.Action, id);
    }
``` 

And after adding a few debug messages, I'm getting a consistent 1128 which is `1128,60,8,0,False,"Unable to use that action at this time."` in the [Log-Message sheet](https://github.com/xivapi/ffxiv-datamining/blob/master/csv/LogMessage.csv). Since Mooch 2 is "gray" until you have caught a fish it's usable on, I guess this makes sense. I also checked what code I'd get when Mooch 2 had been used and was on the timer, in that case I got a 582 `582,60,256,0,False,"Not yet ready."`

Creating a separate method that checks for 1128 instead of 0 does seem to have the setting working as expected.

**Comment by InitialDet** [comment]
Found the issue, will be fixing it on the next update.

**Comment by InitialDet** [comment]
For reasons i have no idea why, IsRecastTimerActive wasnt checking the cooldown for mooch2. 
So i just used a method just like the one i used for the cordials.

```c#
var group = GetRecastGroups(id, actionType);
        
if (group == -1) // Im assuming -1 recast group has no CD
    return false; 
        
var recastDetail = _actionManager->GetRecastGroupDetail(group);
        
return recastDetail->Total - recastDetail->Elapsed > 0;
```

**Comment by InitialDet** [comment]
Should be fixed on 3.0.0.0
